### PR TITLE
Updated fooof.md

### PR DESCRIPTION
**Pull request by ZeyangYu**
Hello,

Line 54 was doing "original"-"fractal". But according to the FOOOF paper, the python implementation, and the current fieldtrip implementation. The subtraction should be operated on log-spectra, thus proposing changing Ln54 to : 
cfg.operation     = '10^(log10(x2)-log10(x1))';%'x2-x1';

Best,
Zeyang

### CODE REVIEW

**Comment by ZeyangYu** [comment]
Reference to fieldtrip function "ft_freqanalysis" Ln 53-54
```
%                       it's expressed as
%                       10^(log10(fooof)-log10(fooof_aperiodic))
``` 

If I have misunderstood the definition of "oscillatory" in this simulated fooof example, please let me know.

**Comment by schoffelen** [comment]
@ZeyangYu I don't understand well why the subtraction SHOULD be done on the log-spectra. I would say that both is possible. I did not provide this answer, and I don't care either way. Perhaps @ruiliu-psy can chime in. Note that examples just serve as examples, they don't serve as a ground truth.

**Comment by ruiliu-psy** [comment]
> @ZeyangYu I don't understand well why the subtraction SHOULD be done on the log-spectra. I would say that both is possible. I did not provide this answer, and I don't care either way. Perhaps @ruiliu-psy can chime in. Note that examples just serve as examples, they don't serve as a ground truth.

Hi both,

The proposed change seems not returning the spectrum in a correct scale.
zeyang

As for the ref, cfg.output = 'fooof_peaks' indeed returns '10^(log10(fooof)-log10(fooof_aperiodic))'. Note that this operation is not based on the original spectrum (the red and yellow lines lay on top of each other.). 
fooofpeaks

If you take the simulated data as your 'ground truth' here, for instance, you could try exclude all oscillatory components in the simulated data (i.e. O = 0;), the 'fooof_peaks' still returns very large power which doesn't look right to me. So in my own study, I don't use this approach.
I also discussed the log-transformation with other FOOOF users and we don't quite understand the rationale behind. You might want to ask the authors.

**Comment by ZeyangYu** [comment]
@ruiliu-psy @schoffelen Hi both! Thanks a lot for your responses. 

First off, let me clarify why I brought this up. Referencing the original Donoghue et al. 2020 paper: in the methods, they specifically pointed out "the algorithm [to recurrently fit gaussian peaks] operates on PSDs in _semilog_-power space...." (see the screenshot below).
image

This is corroborated in the FOOOF python implementation, where we can find the PSD transformed into log10 space before used to determine the rhythmic component:
https://github.com/fooof-tools/fooof/blob/8788431cc42284d7178a22a474bb0fb5a0457f29/fooof/objs/fit.py#L1332

Secondly, as @ruiliu-psy pointed out, in the current fieldtrip implementation of also expresses the "fooof_peaks" as '10^(log10(fooof)-log10(fooof_aperiodic))'
image
https://github.com/fieldtrip/fieldtrip/blob/298c7c2cbd4f2ed5f6178e075f324eb1ec7d3713/ft_freqanalysis.m#L54C37-L54C37

However, the point is well taken that once we've determined the aperiodic component $L_{linear}(f)$ from the orginal $PSD_{linear}(f)$, there is nothing preventing one to compute 
$R_{linear}(f) := PSD_{linear}(f) - L_{linear}(f)$ 
and define it the **rhythmic component**. And it is thus okay for $R(f) \not\equiv \verb|fooof_peaks|$ . 

However, I believe it would be helpful from the "example" point of view to point out this difference for new users.

**Comment by ruiliu-psy** [comment]
Hi Zeyang,

Thank you for digging into this issue! 

Sorry, I get lost after "This is corroborated in the FOOOF python implementation, ..." and am not sure what outcome you want to get. I'd suggest you modifying the example/the implementation as you intended and provide some plots, so that viewers might have a clearer view what this PR is for.

I appended a guideline below, as an example how to initiate an issue (more or less the same as submit a PR). Note that JM as well as Robert have a heavy load of maintaining. It would be helpful that we describe the problem and potential solution in a more organised and legible way.

> **Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. For example: It would be more efficient if [...], or I would like to do [...] to analyze my data.

**Describe the solution you'd like**
A clear and concise description of what you want to happen. How can you contribute to the implementation of your preferred solution?

**Describe alternatives you've considered**
A clear and concise description of any alternative solutions or features you've considered.

**Additional context**
Add any other context or screenshots about the feature request here.

Thanks,
Rui

**Comment by ZeyangYu** [comment]
Hi Rui @ruiliu-psy ,

Great point. Let me restructure as following.

### Problem
- In this example demonstrating FOOOF on simulated data, the so called $oscillatory$ component is computed as `cfg.operation     = 'x2-x1';`. This implies the extracted oscillatory component follows: 
             $oscillatory\ component_{(lin-f, lin-p)}^{[ ver\ 1]} := full\ spectrum_{(lin-f, lin-p)} - fractal\ component_{(lin-f, lin-p)}$, where ($lin-f,lin-p$) denotes linear-frequency, linear-power space. For simplicity, let's assume all further metion of spectral component is by-default linear in frequency and power, unless otherwise indicated.
- While this is technically true [see below for justifactions], it is worthwhile pointing out to new users (supposed target audience of this example) that such expression is not in accordance with the original FOOOF implementation by Donoghue et. al. 2020. 

- Briefly, in the original paper,  $oscillatory$ component is instead computed through the semilog-power (linear frequency, log10-power) space and transformed back into linear-linear space. Such that  $oscillatory\ component^{[ver\ 2]} := 10^{log10(full\ spectrum) - log10(fractal\ component)} = (full\ spectrum)/(fractal\ component) $

- Note, it's easy to see that these two implementaions are not equivalent, as $oscillatory\ component^{[ver\ 2]} := (full\ spectrum)/(fractal\ component)$, while $oscillatory\ component^{[ver\ 1]} := (full\ spectrum) - (fractal\ component) $

- However, both expression have their merits and correponding physiological applications. 

### Proposed Solution
- added a section in this example to demonstrate this distinction: i.e. when `cfg.operation     = 'x2./x1'; % equivalent to 10.^(log10(x2)-log10(x1))`. 
- updated the result image to reflect the modifications
example_2

### Additional Context 
- please see my comment above (https://github.com/fieldtrip/website/pull/692#issuecomment-1679279584) for justifications on the difference between the two expressions of the oscillatory component.

**Comment by ruiliu-psy** [comment]
Thanks @ZeyangYu! Did you get the plot by this example code exactly [6485e66](https://github.com/fieldtrip/website/pull/692/commits/6485e6656401264bdd0caf8ee6a7d57003c600e8)? I ran your code, but the oscillatory spectrum is still on a different scale comparing to the original and fractal.

Screenshot 2023-08-15 at 22 09 10

**Comment by ZeyangYu** [comment]
Thank you @ruiliu-psy! I forgot I had incorporated a different scaling of spectrum in my fieldtrip repository (completely un-related to fooof). I just pulled and tested with https://github.com/fieldtrip/website/pull/692/commits/6485e6656401264bdd0caf8ee6a7d57003c600e8 and arrived at the same plots as you showed. Committed the new image.
example_2

Regarding your comment that:
> oscillatory spectrum is still on a different scale comparing to the original and fractal.

This is some what expected since what is plotted on the left is `log(spectrum - fractal)` while on the right `log(spectum./fractal)`. The former is perhaps more in-line with the perspective of "spectral decomposition", i.e. sums back to the original spectrum. In this case, a requirement of "on the same scale" does make sense. 

In the _linear space_, the latter resembles more of a "normalization" approace. Nevertheless, it also finds common analogies in electrophysiology, e.g. relative change between a task period and a baseline period is expressed as `spectrum_task/spectrum_baseline`. Therefore, the oscillatory component computed this way is not expected to be in the scale/unit of power spectrum (and should not be interpreted as power). 

That being said, if we consider the semi-log space, we can easily show the latter is indeed a "decomposition" of the log-spectrum: i.e. log(oscillatory) = log(spectrum) - log(fractal). 

In the end, it becomes a question of decomposing the linea-space spectrum vs log-space spectrum. So as both you and @schoffelen have pointed out, perhaps both approaches can find their applications. And it was not a black and white scenario at all. I felt it could be helpful for the audience of this tutorial to be aware these differences.

p.s. I am fairly new to FOOOF (yet excited about it) so please don't hesitate to point out my misunderstandings. I am eager to learn from you and other veteran FOOOF users about best-practices and caveats. 

Thank you very much again for your time!

**Comment by schoffelen** [comment]
thanks @ZeyangYu and @ruiliu-psy for the clarification, discussion, and proposed changes. I think that this can be merged, feel free to reopen if you think that things need to be changed a bit more.

**Comment by ruiliu-psy** [comment]
> thanks @ZeyangYu and @ruiliu-psy for the clarification, discussion, and proposed changes. I think that this can be merged, feel free to reopen if you think that things need to be changed a bit more.

Hi JM, I think it might be better to make a separate note behind the example, such as [this](https://www.fieldtriptoolbox.org/example/irasa/#updates-to-the-irasa-implementation). 
My concern was that the proposed operation doesn't really return a 'power' spectrum, which leads confusion to many FOOOF users. Also, if one somehow want the "x2./x1", they can just say cfg.output = 'fooof_peaks';
I'm having a deadline atm, and I can do the edition next week :)

**Comment by schoffelen** [comment]
Ah, OK, thanks for the clarification @ruiliu-psy I suggest that you take it from the merged PR and settle on something updated. No hurry as far as I am concerned.

**Comment by ZeyangYu** [comment]
Thank you both very much!
________________________________

**Comment by ruiliu-psy** [comment]
I tried my best to understand this PR, but I'm still don't quite understand the rational behind. Here are two possibilities that I could think about, please correct me if I'm wrong, @ZeyangYu .

1. You might get confused among the paper and the implementation itself. In FOOOF implementation, the output is already transferred back to the raw power spectrum space, tough the model fit was done in log-space as the paper presented. In this case, I'm not sure if it's really a good idea to compute 'x1./x2' spectrum.

2. Alternatively, you might have some needs for your own analysis, so that you indeed want to use the 'x1./x2' spectrum. It's also fine, but I would suggest customising your own pipeline. I do not recommend change the webpage, as this is not a common approach, other viewers (such as me) might have a difficulty to see your picture.

@schoffelen If @ZeyangYu agreed, I would suggest not to merge this PR and close this case.

**Comment by ZeyangYu** [comment]
Hi Rui,

Really appreciate your time to look into this. Here are my thoughts to your comments.


  1.  "In FOOOF implementation, the output is already transferred back to the raw power spectrum space, tough the model fit was done in log-space as the paper presented. "

     *   Thank you for pointing this out. This is exactly what I wish to reflect/include on the webpage for this PR. Since the audience of this webpage is very likely new FOOOF users, I believe it is helpful to clarify to the new users that 1) the current FOOOF implementation is actually a "decomposition of log spectrum"; and 2) while there is nothing preventing transforming back to linear space and report "x1-x2", the reported subtraction is really never a part of the underlying fitting process, yet "x1./x2" is.
  2.  "I'm not sure if it's really a good idea to compute 'x1./x2' spectrum."
     *   'x1./x2' is a common normalization method for TFR (termed "relative" in fieldtrip ft_freqbaseline.m [https://github.com/fieldtrip/fieldtrip/blob/master/ft_freqbaseline.m]), as it helps accounting for 1/f in comparing TFRs under different conditions.
     *   my intention to show the 'x1./x2' spectrum was not to necessarily to promote reporting FOOOF decomposed spectrum this way. I would certainly love to use 'x1-x2' worry-free, but echoing the point above, I struggle to reconcile with what is actually being fitted.

In short, this PR discussion has already been greatly helpful for my understanding in FOOOF. I have no problem with removing the associated plot/code if both you and @schoffelen<https://github.com/schoffelen> think it is confusing. However, I would request keeping/adding some clarifying language regarding point 1.

Many thanks!
Best,
Zeyang

     *


________________________________

**Comment by ruiliu-psy** [comment]
@schoffelen I'm not sure I'm able to understand Zeyang's concerns completely. 
I don't have strong opinion on this. Just for your reference, 
1. Consistent with the algorithm presented in the original paper, process_fooof fits models semilog space. 
2. [process_fooof](external/brainstorm/process_fooof.m) returns the output in the linear space (see line 287: fg(chan).ap_fit = 10.^ap_fit;).  
Thus, I don't spot anything wrong with (mixed - aperiodic) as an estimation of periodic spectrum. Hope it's clear once for all :)
